poethepoet/context.py

```diff
@@ -105,11 +106,15 @@ def get_executor(
     ) -> "PoeExecutor":
         from .executor import PoeExecutor
 
+        cwd_option = task_options.get("cwd", ".")
+        if cwd_option == '$exec_cwd':
+            cwd_option = os.getcwd()
+
         return PoeExecutor.get(
             invocation=invocation,
             context=self,
             env=env,
-            working_dir=self.project_dir / task_options.get("cwd", "."),
+            working_dir=self.project_dir / cwd_option,
```

Comment:
The logical flow here isn't quite right; I if `cwd_option` is set from `os.getcwd()`  then it doesn't make sense to join it with `self.project_dir`.

Reply:
Functionally, as [documented in `pathlib`](https://docs.python.org/3/library/pathlib.html#operators), this should be fine - if the right operand is an absolute path, the left operand gets discarded. The result of `os.getcwd()` will surely always be an absolute path, at least in my mind it wouldn't make much sense if it weren't - although the docs don't explicitly say that.

That said, I can rewrite it to be a little more explicit, if you wish.

Reply:
Understood, it's probably work fine as it is... still could be clearer/more correct.

Reply:
Made this more explicit, how does it look to you?